lib/src/gutter.dart

```diff
@@ -1,54 +1,123 @@
-import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
 
-import 'package:flutter_gutter/flutter_gutter.dart';
-
-/// A gap twice the standard gutter size according to Material Design's
-/// breakpoints system.
-class GutterLarge extends StatelessWidget {
-  const GutterLarge({super.key});
-
-  @override
-  Widget build(BuildContext context) => Gap(size: context.gutterLarge);
-}
+import '../flutter_gutter.dart';
 
 /// A gap of the standard gutter size according to Material Design's breakpoints
 /// system.
+/// Creates a `SizedBox` with a height of size corresponding to the layout axis
+/// of the first `Scrollable`, `Row` or `Column` above this widget.
 class Gutter extends StatelessWidget {
-  const Gutter({super.key});
+  /// Creates a new [Gutter] widget.
+  const Gutter({
+    super.key,
+    this.size,
+    this.type = GutterType.medium,
+    this.scaleFactor,
+  });
+
+  /// Creates a new [Gutter] widget with a tiny size.
+  const Gutter.tiny({super.key})
+      : size = null,
+        type = GutterType.tiny,
+        scaleFactor = null;
+
+  /// Creates a new [Gutter] widget with a small size.
+  const Gutter.small({super.key})
+      : size = null,
+        type = GutterType.small,
+        scaleFactor = null;
+
+  /// Creates a new [Gutter] widget with a medium size.
+  const Gutter.medium({super.key})
+      : size = null,
+        type = GutterType.medium,
+        scaleFactor = null;
+
+  /// Creates a new [Gutter] widget with a large size.
+  const Gutter.large({super.key})
+      : size = null,
+        type = GutterType.large,
+        scaleFactor = null;
+
+  /// Creates a new [Gutter] widget with an extra large size.
+  const Gutter.extraLarge({super.key})
+      : size = null,
+        type = GutterType.extraLarge,
+        scaleFactor = null;
+
+  /// The type of gutter to create.
+  final GutterType type;
+
+  /// The size of the gap.
+  final double? size;
+
+  /// The scale factor to apply to the gutter size.
+  final double? scaleFactor;
 
   @override
-  Widget build(BuildContext context) => Gap(size: context.gutter);
+  Widget build(BuildContext context) {
+    return AxisAware(
+      key: key,
+      builder: (BuildContext context, Orientation orientation) {
```

Comment:
Tiny nit: lets drop the types here for consistency with other anon functions in builders.

Reply:
I've enabled strong types everywhere because it is easier to read when using sources. The flutter team recommends this.